components/raftstore/src/store/fsm/store.rs

```diff
@@ -292,16 +292,21 @@ where
 {
     fn notify(&self, apply_res: Vec<ApplyRes<EK::Snapshot>>) {
         for r in apply_res {
-            self.router.try_send(
-                r.region_id,
+            let region_id = r.region_id;
+            if let Err(e) = self.router.force_send(
+                region_id,
                 PeerMsg::ApplyRes {
                     res: ApplyTaskRes::Apply(r),
                 },
-            );
+            ) {
+                error!("failed to send apply result"; "region_id" => region_id, "err" => ?e);
```

Comment:
What if it's shutting down?

Reply:
Already checked here https://github.com/tikv/tikv/blob/1e13ddf3bf12c00afb4d049d05978fecae9a6067/components/batch-system/src/router.rs#L263